src/languages/colour.ts

```diff
@@ -0,0 +1,81 @@
+import { ExtensionContext, l10n, Range, window, workspace, WorkspaceEdit } from "vscode";
+import { instance } from "../instantiate";
+
+const NEW_LINE_NUMBERS = [10, 13];
+
+export function initialiseColourChecker(context: ExtensionContext) {
+  context.subscriptions.push(
+    workspace.onDidOpenTextDocument(async (document) => {
+      if (document.uri.scheme === `member` && !document.isClosed) {
+        const content = document.getText();
+        let doWork = hasInvalidCharacters(content);
+
+        if (doWork) {
+          const shouldFix = await askUserToStart();
+
+          if (shouldFix) {
+            const fixedContent = replaceInvalidCharacters(content);
+            const edit = new WorkspaceEdit();
+            edit.replace(document.uri, new Range(0, 0, document.lineCount, 0), fixedContent);
+            workspace.applyEdit(edit);
+          }
+        }
+      }
+    })
+  )
+}
+
+export function hasInvalidCharacters(content: string) {
+  for (let i = 0; i < content.length; i++) {
+    if (shouldReplaceCharCode(content.charCodeAt(i))) {
+      return true;
+    }
+  }
+  return false;
+}
+
+function shouldReplaceCharCode(charCode: number) {
+  if ((charCode < 32 && !NEW_LINE_NUMBERS.includes(charCode)) || (charCode >= 128 && charCode <= 157)) {
+    return true;
+  }
+  return false;
+}
+
+async function askUserToStart() {
+  const config = instance.getConfig()
+
+  if (config?.autoFixInvalidCharacters) {
+    return true;
+  }
+
+  const always = l10n.t(`Always`);
+  const no = l10n.t(`No`);
+  
+  const chosen = await window.showInformationMessage(
+    l10n.t(`seuColours.warning`), 
```

Comment:
This was lost after migration to l10n
```suggestion
    l10n.t("This member contains invalid characters. Would you like to clean it up?"), 
```